Automattic/wp-calypso
# ExPlat: Implement A/A tests

#### Changes proposed in this Pull Request

**This is a risky PR:** We are running a fairly large amount of new code in production in some fairly sensitive places.

This PR adds three A/A tests for the in-progress ExPlatJsClient:

* `explat_test_aa_calypso_boot`: `loadExperimentAssignment` in general calypso boot, run if logged in (so it won't run on the server), multiple calls with `useExperiment` and `dangerousGetExperimentAssignment` higher up.
* `explat_test_aa_calypso_signup`: `loadExperimentAssignment` in Calypso signup `index.web.ts`, with a `dangerousGetExperimentAssignment` higher up.
* `explat_test_aa_weekly_calypso_next_client_*`: Experiment component mirroring the existing evergreen A/A.

It aslo applies some minor fixes:
- Using `export type` and `import type`.
- Exporting `getTracksLoadPromise`.
- Changes `initializeAnonIdPromise` from a `const` to a `let` to fix an edge case.

#### Testing instructions



* Visit signed-in Calypso: ensure the `boot` experiment request is there.
* Visit signed-out Calypso: ensure the `signup` experiment request is there.
* Visit the Calypso Pages section: ensure the new evergreen aa experiment request is in there.
* Ensure there are no unusual console logged errors from the above.





Related to 585-gh-Automattic/experimentation-platform
